shared/api.ts

```diff
@@ -0,0 +1,161 @@
+import {request} from './request';
+import Storage, {API_CACHED_STORAGE_KEYS} from './storage';
+import {COMMON_PROPS_KEY} from './commonprops/commonProps';
+import {INITIAL_CONFIG} from './InitialConfig';
+
+export const API_URLS = {
+  issuersList: {
+    method: 'GET',
+    buildURL: (): `/${string}` => '/residentmobileapp/issuers',
+  },
+  issuerConfig: {
+    method: 'GET',
+    buildURL: (issuerId: string): `/${string}` =>
+      `/residentmobileapp/issuers/${issuerId}`,
+  },
+  allProperties: {
+    method: 'GET',
+    buildURL: (): `/${string}` => '/residentmobileapp/allProperties',
+  },
+};
+
+export const API = {
+  fetchIssuers: async () => {
+    const defaultIssuer = {
+      id: 'UIN, VID, AID',
+      displayName: 'UIN, VID, AID',
+    };
+
+    const response = await request(
+      API_URLS.issuersList.method,
+      API_URLS.issuersList.buildURL(),
+    );
+    return [defaultIssuer, ...(response.response.issuers || [])];
+  },
+
+  fetchIssuerConfig: async (issuerId: string) => {
+    const response = await request(
+      API_URLS.issuerConfig.method,
+      API_URLS.issuerConfig.buildURL(issuerId),
+    );
+    return response.response;
+  },
+
+  fetchAllProperties: async () => {
+    const response = await request(
+      API_URLS.allProperties.method,
+      API_URLS.allProperties.buildURL(),
+    );
+    return response.response;
+  },
+};
+
+export const CACHED_API = {
+  fetchIssuers: () =>
+    catchAPIMethod({
+      cacheKey: API_CACHED_STORAGE_KEYS.fetchIssuers,
+      fetchCall: API.fetchIssuers,
+    }),
+
+  fetchIssuerConfig: (issuerId: string) =>
+    catchAPIMethod({
+      cacheKey: API_CACHED_STORAGE_KEYS.fetchIssuerConfig(issuerId),
+      fetchCall: API.fetchIssuerConfig.bind(null, issuerId),
+    }),
+
+  getAllProperties: () =>
+    catchAPIMethod({
+      isCachePreferred: true,
+      cacheKey: COMMON_PROPS_KEY,
+      fetchCall: API.fetchAllProperties,
+      onErrorHardCodedValue: INITIAL_CONFIG.allProperties,
+    }),
+};
+
+interface cacheAPIMethodProps {
+  isCachePreferred?: boolean;
+  cacheKey: string;
+  fetchCall: (...props: any) => Promise<any>;
+  onErrorHardCodedValue?: any;
+}
+
+async function catchAPIMethod({
```

Comment:
Do we need to use keyword `method` in function name?
and it's catchAPI or cacheAPI?